docs/source/json-api/index.rst

```diff
@@ -1004,13 +1004,29 @@ HTTP Response with Unknown Parties Warning
         }
       ],
       "warnings": {
-        "unknownParties": [
-          "Erin"
-        ]
+        "unknownParties": ["Erin"]
       },
       "status": 200
     }
 
+BadRequest(400) Response with Unknown Parties Warning
```

Comment:
I'm not convinced this is the right behaviour. I assume the intent is to bring it line with the `query` endpoint. However, parties are a runtime thing whereas templates are a compile time thing. Thus, asking for a list of templates that happens to be empty is very likely a programming error. In contrast, asking for a list of parties where none of them can be resolved seems like more usual business to me. Thus, changing the behaviour depending on whether at least one party could be resolved is the wrong thing to do in my opinion. (Asking for an empty list of parties should still be an error though since that's clearly a programming error.)

Reply:
@hurryabit yes, that is the intent.

In my experience if you are fetching something by ID (template ID or party ID) and there is no entity with that ID, -- this is an error. It is a common practice in RESTful APIs to return 40x status in this case.... we are not RESTful though....

In any case I think we should return an error, even if one of the IDs does not exist, but I remember you had a strong opinion about non-existent template IDs.

Up to you, it is easy to change this error to OK(200) for the party lookup by ID.

Reply:
PR: #5456